### ISSUE TITLE
Validation split leaks: 12 transpositions of a chunk land on both sides

### ISSUE THREAD

**Issue by metzgerdev** [metzgerdev/midi_gpt#2]
`train_notes.py:103` splits the dataset *after* key augmentation, so transpositions of the same musical chunk end up in both train and validation. Validation loss is therefore close to training loss, and "best val" checkpoint selection is close to meaningless.

## Cause

`mine_corpus.py:122-127` writes each 4-bar chunk twelve times, once per key:

```
SO_UKG_140_bass_5th_reese_Bm_479d8d2c6f__0_k0.npz
SO_UKG_140_bass_5th_reese_Bm_479d8d2c6f__0_k1.npz
...                                        __0_k11.npz
```

`NoteCorpus` loads all of them into one flat list, then:

```python
train_set, val_set = random_split(dataset, [len(dataset) - n_val, n_val],
                                  generator=torch.Generator().manual_seed(0))
```

The bass corpus is 4,248 files = **354 distinct chunks x 12 keys**. With a 15% split, essentially every validation item has 11 near-duplicates in training — the same rhythm, the same contour, the same conditioning, transposed. This is the standard near-duplicate contamination problem from text pretraining.

## Impact

- Reported val loss is not a held-out measurement.
- `if val < best` selects checkpoints on a metric that cannot detect overfitting.
- No signal exists today for whether 150 epochs is too many or too few.

## Fix

Group the split by chunk identity before augmenting. The filenames already encode it — `<stem>_<digest>__<chunk>_k<shift>.npz` — so partition on `(digest, chunk)` and assign all twelve keys of a chunk to the same side.

Worth pairing with a second change: the training objective is class-weighted (`train_notes.py:115-117`), so the logged number is not a perplexity and is not comparable across configurations. Logging unweighted held-out cross-entropy alongside it would give a real metric to select on.
